### PR TITLE
Metrics

### LINKED ISSUE #20: Expand metrics provided in report
See https://github.com/JaysGitLab/cs-5666-scatt-gp/wiki/Metrics for possible other metrics.

### PR DESCRIPTION
**Pull request by haietza**
Closes #20 
Add counts for remaining items
* stage
  * variables
  * lists
  * scripts
  * scriptComments
  * sounds
  * costumes
* children (sprites)
  * variables
  * lists
  * scripts
  * scriptComments
  * sounds
  * costumes

### CODE REVIEW

**Comment by KaraBeason** [comment]
@haietza I get a null pointer exception when running tests.  Screenshot:
nullptrex

File: Submission.java [hunk 1]
```diff
@@ -181,23 +258,95 @@ private JSONArray getSprites()
...
+     * @param item 
+     * @param spriteName 
+     * @return count 
+     */
+    private int getCountForSprite(String item, String spriteName)
+    {
         JSONArray sprites = getSprites();
-        JSONArray scripts = new JSONArray();
+        JSONArray items = new JSONArray();
         for (int i = 0; i < sprites.size(); i++)
```
Comment (KaraBeason) [inline Submission.java]: Is there any way to make this function or its variables more descriptive?  Or is it intentionally generic?  Looking at it right now I can't infer what its intended purpose is

Comment (haietza) [inline Submission.java]: @KaraBeason So, this was code I refactored out of all the countForSprite methods. Instead of repeating this code over and over, I pulled it out so the other functions call this and pass it the name of the sprite to get the count for, and the JSON attribute name of the item to count (costumes, scripts, etc). What do you think would be a more descriptive name for the function/variables? It's kind of a generic function, with the specificity being passed in as parameters, so I guess that's why I named it kind of generic - but if you have a suggestion for something that is more self-documenting, I'm happy to change it!

Also, I was passing the wrong array element to one of the tests, which was causing the null pointer exception above. I just pushed a change - take a look and let me know if that fixes it on your end.

Thanks so much!

Comment (KaraBeason) [inline Submission.java]: @haietza That makes sense.   Maybe you could incorporate that into the method's javadoc comment, something like "this is a generic helper function used by the countForSprite methods"?  Idk, what do you think?

File: SubmissionTest.java [hunk 2]
```diff
@@ -292,39 +282,267 @@ public void testGetScriptCountValid()
...
+     */
+    @Test
+    public void testGetScriptCountForStageInvalid()
+    {
+        setUpSubmission(submissions[1]);
+
+        int unexpected = 2;
+        int actual = submissions[1].getScriptCountForStage();
+        assertFalse("should be false", unexpected == actual);
+    }
```
Comment (KaraBeason) [inline SubmissionTest.java]: I'm pretty sure this unit test is not necessary.  It seems trivial to test that a function's output is not equal to an arbitrary number.

File: SubmissionTest.java [hunk 3]
```diff
@@ -292,39 +282,267 @@ public void testGetScriptCountValid()
...
+     */
+    @Test
+    public void testGetVariableCountForStageInvalid()
+    {
+        setUpSubmission(submissions[1]);
+
+        int unexpected = 4;
+        int actual = submissions[1].getVariableCountForStage();
+        assertFalse("should be false", unexpected == actual);
+    }
```
Comment (KaraBeason) [inline SubmissionTest.java]: same deal here

File: Submission.java (hunk 1 above)
Comment (haietza) [inline Submission.java]: @KaraBeason Good call. I added the comment to help clarify and changed the 'item' parameter name to 'attribute' so hopefully that makes it a bit more clear. 

I had the same thought about the invalid tests - somewhere along the line I started testing empty attributes, which I think is what I should have been doing for the 'invalid' testing of these functions all along. I went in and removed the useless invalid tests and added tests for empty values where they weren't already implemented. 

Take a look and let me know if you see anything else. Thanks much!

File: Submission.java [hunk 4]
```diff
@@ -133,6 +133,83 @@ public int getScriptCount()
...
+        if (items != null)
+        {
+            return (int) items.size();
+        }
+        return 0;
+    }
+
+    /**
      * Get array of sprites.
      * Unchecked warnings are suppresed because JSONArray does not
```
Comment (KaraBeason) [inline Submission.java]: I just realized there is a typo in suppressed and it is totally mine haha.  If you wouldn't mind fixing it I would appreciate it.  Sorry about that.

File: Submission.java (hunk 1 above)
Comment (KaraBeason) [inline Submission.java]: Everything looks great!  Thanks for all your hard work!

File: Submission.java (hunk 4 above)
Comment (haietza) [inline Submission.java]: @KaraBeason I'm sorry - this one was actually me! Too bad checkstyle doesn't check spelling too! :) Thanks for reviewing this!
